daemon/api_test.go

```diff
@@ -134,8 +130,7 @@ oem: {store: {id: %q}}
 }
 
 func (s *apiSuite) TestPackageInfoOneIntegration(c *check.C) {
-	d := New()
-	d.addRoutes()
+	newTestDaemon()
```

Comment:
The branch looks fine as it's not really making matters any worse, but the lack of context here indicates some disturbing use of global state. How come this creates something that is dropped on the floor, yet it somehow manages to get picked up by the logic below? And finalized somehow?

I really hope we can get to a point where there's no global state, and the setup an termination are clear and in context.